clients/src/main/java/org/apache/kafka/common/security/auth/SaslExtensions.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.security.auth;
+
+import org.apache.kafka.common.utils.Utils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class SaslExtensions {
+    protected Map<String, String> extensionMap;
+    protected String separator;
+
+    protected SaslExtensions() {
+        separator = ",";
+        extensionMap = new HashMap<>();
+    }
+
+    public SaslExtensions(String extensions, String separator) {
+        this(Utils.parseMap(extensions, "=", ","), separator);
+    }
+
+    public SaslExtensions(Map<String, String> extensionMap, String separator) {
+        this.extensionMap = Collections.unmodifiableMap(new HashMap<>(extensionMap));
+        this.separator = separator;
+    }
+
+    public String extensionValue(String name) {
+        return extensionMap.get(name);
+    }
+
+    public Set<String> extensionNames() {
+        return extensionMap.keySet();
+    }
+
+    public boolean isEmpty() {
+        return extensionMap.isEmpty();
+    }
+
+    /*
+        Returns a HashMap copy of the extensions map
+     */
+    public Map<String, String> extensionsMap() {
```

Comment:
Probably no need to make a copy since the map is unmodifiable; Javadoc can say it returns an unmodifiable Map view of the extensions.  This raises the question of whether all of the Map-related methods isEmpty(), extensionNames(), extensionEntries(), and extensionValue() are actually needed.  If returning the map is cheap -- which it would be -- then all of those methods feel like clutter rather than good API additions.  What do you think?